src/main/java/org/terasology/ligthandshadow/componentsystem/controllers/TeleporterSystem.java

```diff
@@ -100,6 +101,10 @@ private boolean isProperTeamSize(EntityRef teleporter, EntityRef player) {
             }
         }
         if (teleporterTeamCount - oppositeTeamCount < maxTeamSizeDifference) {
+            if (teleporterTeamCount >= 0 && oppositeTeamCount >= 1 && !gameStart) {
+                sendTimerEventToClients();
+                gameStart = true;
+            }
```

Comment:
Not sure if this should be in `isProperTeamSize` :thinking: 
This seems to be a more or less unnecessary side effect.
Like not that the logic is unnecessary at all, but rather that it being a side effect here is unnecessary...

Reply:
Wouldn't iterating through the players again to get the count of players be code duplication.

Reply:
And suddenly, keeping this information as state seems more appealing again :see_no_evil: 
@skaldarnar what's your pov on this?

Reply:
So Skal just argued that looping through all players again would not be too much of a concern as we have - in comparison to the total number of entities - comparably few players.
@ahv15 can you maybe put a comment on the overhaul trello card and note to have another look at this as a follow-up?

Reply:
Ok